lesson12/index.js

```diff
@@ -0,0 +1,63 @@
+const ul = document.getElementById('js-lists');
+const fetchButton = document.getElementById('fetchButton');
+
+const displayLoadingImage = () => {
+    const loadingImage = document.createElement('img');
+    loadingImage.id = 'loadingImageId';
+    loadingImage.src = './loading-circle.gif';
+    ul.appendChild(loadingImage);
+}
+
+const removeLoadingImage = () => {
+    const loadingImage = document.getElementById('loadingImageId');
+    loadingImage.remove();
+}
+
+const getData = async (url) => {
+    try {
+        const response = await fetch(url);
+        return await response.json();
+    } catch (error) {
+        throw new Error(error);
+    }
+};
+
+const init = async () => {
+    displayLoadingImage();
+    try {
+        const { data } = await getData("https://myjson.dit.upm.es/api/bins/edwr");
+        createList(data);
+    } catch (e) {
+        errorText(`エラーが発生しました: 詳細 ${e}`);
+    }
+    removeLoadingImage();
+}
+
+const errorText = (text) => {
+    const errorText = document.createElement('p');
+    ul.replaceWith(errorText);
+    errorText.textContent = text;
+}
+
+const createList = (lists) => {
+    const frag = new DocumentFragment();
+    for(const list of lists) {
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        a.href = list.a;
+        const img = document.createElement('img');
+        img.src = list.img;
+        img.alt = list.alt;
+        const text = document.createTextNode(list.text);
+
+        a.appendChild(img);
+        a.appendChild(text);
+        li.appendChild(a);
+        frag.appendChild(li);
+    }
+    ul.appendChild(frag);
+}
+
+fetchButton.addEventListener('click', () => {
+    init();
+});
```

Comment:
Since init() is executed every time a button is clicked, the list displayed will grow indefinitely.

When the button is clicked, it is better to hide it.

Reply:
Thank you for your reviewing!
When the button is clicked, it is removed. 
-> 6c6a888

Reply:
I'm getting an error "Not defined".
<img width="476" alt="スクリーンショット 2021-11-26 3 35 47" src="https://user-images.githubusercontent.com/43788628/143489566-031bc15e-9f8b-4a1e-8250-5b123af8262d.png">